src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityItemBus.java

```diff
@@ -110,6 +124,96 @@ protected ModularUI createUI(EntityPlayer entityPlayer) {
         return builder.bindPlayerInventory(player.inventory, GuiTextures.SLOT, 7 + xOffset, 18 + 18 * rowSize + 12);
     }
 
+    private static void collapseInventorySlotContents(IItemHandlerModifiable inventory) {
+        //stack item stacks with equal items and compounds
+        for (int i = 0; i < inventory.getSlots(); i++) {
+            for (int j = i + 1; j < inventory.getSlots(); j++) {
+                ItemStack stack1 = inventory.getStackInSlot(i);
+                ItemStack stack2 = inventory.getStackInSlot(j);
+                if (!stack1.isEmpty() && ItemStack.areItemsEqual(stack1, stack2) &&
+                        ItemStack.areItemStackTagsEqual(stack1, stack2)) {
+                    int maxStackSize = Math.min(stack1.getMaxStackSize(), inventory.getSlotLimit(i));
+                    int itemsCanAccept = Math.min(stack2.getCount(), maxStackSize - Math.min(stack1.getCount(), maxStackSize));
+                    if (itemsCanAccept > 0) {
+                        stack1.grow(itemsCanAccept);
+                        stack2.shrink(itemsCanAccept);
+                    }
+                }
+            }
+        }
+        ArrayList<ItemStack> inventoryContents = new ArrayList<>();
+        for (int i = 0; i < inventory.getSlots(); i++) {
+            ItemStack itemStack = inventory.getStackInSlot(i);
+            if (!itemStack.isEmpty()) {
+                inventory.setStackInSlot(i, ItemStack.EMPTY);
+                inventoryContents.add(itemStack);
+            }
+        }
+        for (int i = 0; i < inventoryContents.size(); i++) {
+            inventory.setStackInSlot(i, inventoryContents.get(i));
+        }
+
+    }
+
+    @Override
+    public boolean onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing, CuboidRayTraceResult hitResult) {
+        IItemHandlerModifiable inventory = (isExportHatch ? this.getExportItems() : this.getImportItems());
+        if (!playerIn.isSneaking()){
+            boolean isAttached = false;
+            if (isExportHatch ? this.getNotifiedItemOutputList().contains(inventory) : this.getNotifiedItemInputList().contains(inventory)){
```

Comment:
This could just check `this.isAttachedToMultiblock()`, as the notified lists might not necessarily contain the hatch.